engine/runtime-instrument-common/src/main/java/org/enso/interpreter/instrument/command/InvalidateModulesIndexCommand.java

```diff
@@ -50,14 +51,16 @@ public Future<BoxedUnit> executeAsynchronously(RuntimeContext ctx, ExecutionCont
 
             StartBackgroundProcessingJob.startBackgroundJobs(ctx);
             reply(new Runtime$Api$InvalidateModulesIndexResponse(), ctx);
+          } catch (InterruptedException ie) {
+            logger.log(Level.WARNING, "Failed to acquire lock: interrupted", ie);
           } finally {
-            ctx.locking().releaseWriteCompilationLock();
-            logger.log(
-                Level.FINEST,
-                "Kept write compilation lock [{0}] for {1} milliseconds.",
-                new Object[] {
-                  this.getClass().getSimpleName(),
-                  System.currentTimeMillis() - writeCompilationLockTimestamp
+            logLockRelease(
+                logger,
+                "write compilation",
+                writeCompilationLockTimestamp,
+                () -> {
+                  ctx.locking().releaseWriteCompilationLock();
```

Comment:
Now when the lock is in the try block. If the lock acquisition fails and you try to release the unlocked lock, you should get the illegal state exception I believe